### PR TITLE
week-3-ahmed

### PR DESCRIPTION
**Pull request by Ahmedkhafaji**
# Submitting Week 3 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ x] Coding Challenges
- [ x] Interview Questions
- [ x] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,15 +15,46 @@
...
+
+
+
+const fibonacci = (num) => {
+  let fib = [1, 1]
+  for (let i = 2; i < num; i++) {
+    fib.push(fib[i - 2] + fib[i - 1])
+  }
+  return fib 
+}
```
Comment (elysemontano) [inline code-challenges.test.js:53]: Nice work Ahmed!

File: code-challenges.test.js [hunk 2]
```diff
@@ -36,8 +67,54 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
...
+////////////////
+//2nd try
+
+const oddNums = (array) => {
+  return array.filter(value => {
+      if(value % 2 !== 0 && typeof (value) === "number"){
+          return array
+      }
+  }).sort((a, b) => a - b)
+}
```
Comment (elysemontano) [inline code-challenges.test.js:108]: This looks great!  I appreciate you leaving in your first attempt as well so I can see your process.

File: code-challenges.test.js [hunk 3]
```diff
@@ -53,4 +130,29 @@ const numbersToAdd3 = []
...
 // b) Create the function that makes the test pass.
+
+//pusedo code
+//create a function called SumNums
+//create a parameter called array
+//use .map to iterate through the arry
+// apply a condition to return the new array with numbers added from the sum of the old array to the new array
+
+const sumNums = (array) => {    let newArray = []  
+    return array.map((value, index) => {        if(index > 0){            return newArray = value + newArray        }else {            return newArray = value        }    })}
```
Comment (elysemontano) [inline code-challenges.test.js:156]: Your logic looks great, just make sure to watch how this is formatted and indentation.

File: interview-questions.md [hunk 4]
```diff
@@ -9,41 +9,41 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render() method is required in a React class component because it can run the code without having to refresh the page.
 
-  Researched answer:
+  Researched answer: lifecycle method has 3 steps. mounting, updating, and unmounting. the mounting is important because it has the constructor, render, and methods.
```
Comment (elysemontano) [inline interview-questions.md:30]: Nicely done!

File: interview-questions.md [hunk 5]
```diff
@@ -9,41 +9,41 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: when yarn is ran, it is basically testing framework to check if the codebase is working or not. 
 
-  Researched answer:
+  Researched answer: yarn is javascript package manager and the package manager is to have code so when yarn start is ran, it opens a server and runs the code.
```
Comment (elysemontano) [inline interview-questions.md:46]: Yarn will manage all of the node modules by recognizing the dependancies in the package.json file.

**Review by elysemontano — APPROVED** [review]
Outstanding work Ahmed!!  You crushed this assessment!  Your logic, pseudocode and tests were solid, interview answers were great and you crushed the React section.  Just make sure to keep an eye on indentation in your code, one problem had some strange formatting happening.  Keep up the great work!

File: code-challenges.test.js (hunk 1 above)
Comment (Ahmedkhafaji) [reply code-challenges.test.js:53]: thank you!!

File: code-challenges.test.js (hunk 2 above)
Comment (Ahmedkhafaji) [reply code-challenges.test.js:108]: yay!!

File: code-challenges.test.js (hunk 3 above)
Comment (Ahmedkhafaji) [reply code-challenges.test.js:156]: got it!!

**Pull request merged**